src/main/scala/li/cil/oc/common/nanomachines/ControllerImpl.scala

```diff
@@ -340,33 +341,33 @@ class ControllerImpl(val player: EntityPlayer) extends Controller with WirelessE
 
   // ----------------------------------------------------------------------- //
 
-  def save(nbt: NBTTagCompound): Unit = configuration.synchronized {
-    nbt.setString("uuid", uuid)
-    nbt.setInteger("port", responsePort)
-    nbt.setDouble("energy", storedEnergy)
-    nbt.setNewCompoundTag("configuration", configuration.save)
+  def saveData(nbt: CompoundNBT): Unit = configuration.synchronized {
+    nbt.putString("uuid", uuid)
+    nbt.putInt("port", responsePort)
+    nbt.putDouble("energy", storedEnergy)
+    nbt.setNewCompoundTag("configuration", configuration.saveData)
   }
 
-  def load(nbt: NBTTagCompound): Unit = configuration.synchronized {
+  def loadData(nbt: CompoundNBT): Unit = configuration.synchronized {
     uuid = nbt.getString("uuid")
-    responsePort = nbt.getInteger("port")
+    responsePort = nbt.getInt("port")
     storedEnergy = nbt.getDouble("energy")
-    configuration.load(nbt.getCompoundTag("configuration"))
+    configuration.loadData(nbt.getCompound("configuration"))
     activeBehaviorsDirty = true
   }
 
   // ----------------------------------------------------------------------- //
 
-  private def isClient = world.isRemote
+  private def isClient = world.isClientSide
 
   private def isServer = !isClient
 
   private def cleanActiveBehaviors(reason: DisableReason): Unit = {
     if (activeBehaviorsDirty) {
       configuration.synchronized(if (activeBehaviorsDirty) {
         val newBehaviors = configuration.behaviors.filter(_.isActive).map(_.behavior)
-        val addedBehaviors = newBehaviors -- activeBehaviors
-        val removedBehaviors = activeBehaviors -- newBehaviors
+        val addedBehaviors = newBehaviors.clone --= activeBehaviors
```

Comment:
It's because scala 2.13.4 doesn't have the difference method `--` anymore, this creates a new list and subtracts the `activeBehaviors` from that in-place. The next line would probably still work but I didn't notice that it uses a Set at the time (nevertheless removing an `IterableOnce` from a mutable `Set` is already deprecated).

Reply:
Got it, thanks for the explanation.